sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -1710,7 +1710,13 @@ private Map<String, String> getRequestHeaders(RequestOptions options, ResourceTy
         }
 
         if (consistencyLevel != null) {
-            headers.put(HttpConstants.HttpHeaders.CONSISTENCY_LEVEL, consistencyLevel.toString());
+            // adding the "x-ms-consistency-level" header with consistency level stricter than the
+            // account's default consistency level in Compute Gateway will result in a 400 Bad Request
+            // even when it is done for resource types / operations where this header should simply be ignored
+            // making the change here to restrict adding the header to when it is relevant.
+            if (operationType.isReadOnlyOperation() && (resourceType.isMasterResource() || resourceType == ResourceType.Document)) {
```

Comment:
so there is no need to populate this header for write operations as those operations will always be on account level consistency? 

and will "Compute Gateway will result in a 400 Bad Request" behavior be changed in compute in future?

Reply:
>and will "Compute Gateway will result in a 400 Bad Request" behavior be changed in compute in future?
No idea - haven't discussed it with them yet. Given it is in place for months and no custoemr complained  I doubt this will be relaxed on Compute

Reply:
And correct - for write operations any consistency setting is completely irrelevant because service will always use the acocunt-level consistency anyway